src/context.cc

```diff
@@ -331,9 +331,24 @@ bool ContextBase::onStart(std::shared_ptr<PluginBase> plugin) {
 }
 
 bool ContextBase::onConfigure(std::shared_ptr<PluginBase> plugin) {
-  if (isFailed() || !wasm_->on_configure_) {
+  if (isFailed()) {
     return true;
   }
+
+  // on_context_create is yet to be executed for all the root contexts except the first one
+  if (!in_vm_context_created_ && wasm_->on_context_create_) {
+    DeferAfterCallActions actions(this);
+    wasm_->on_context_create_(this, id_, 0);
+  }
+
+  // NB: If no on_context_create function is registered the in-VM SDK is responsible for
+  // managing any required in-VM state.
+  in_vm_context_created_ = true;
+
+  if (!wasm_->on_configure_) {
+    return true;
+  }
+
   DeferAfterCallActions actions(this);
```

Comment:
I think you need to make sure that this isn't constructed twice (in the added check and here), i.e. if `onCreate` wasn't called before calling `onConfigure`, then it's going to created and queue this twice.

I think you could see it if you log `proxy_on_plugin_configure` callbacks. Could you please test?

Reply:
I now understand what to do, but what I'm a little bit confused about is that 
- when exactly re-entrant happens
- why `DeferAfterCallActions` prevents the second time execution of the call (in this case `on_context_create_` and `on_configure_` )
    - It occurs to me that `DeferAfterCallActions` just ensures the execution of functions added through `addAfterVmCallAction` which have nothing to do with the these VM calls.

sorry if i missed something and for this naive questions in advance!

Reply:
I looked over the PRs https://github.com/envoyproxy/envoy-wasm/pull/432 https://github.com/envoyproxy/envoy-wasm/pull/597 but I still don't get it

Reply:
Sorry, I misremembered the implementation details. I think this is fine.

Could you please verify that your original test case doesn't crash anymore, and works as expected? I'll merge it once I hear back from you.

Reply:
No problem :+1: Thanks!